### PR TITLE
Em 1904

### PR DESCRIPTION
**Pull request by pawelkaczmarekuk**

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -1,2 +1,5 @@
 SERVER_PORT=8080
+DB_USERNAME=ccd
```
Comment (linusnorton) [inline .env]: what's this for?

File: docker-compose-dependencies.yml [hunk 2]
```diff
@@ -70,30 +77,30 @@ services:
...
-      - DOCMOSIS_ACCESS_KEY=${DOCMOSIS_ACCESS_KEY}
-    depends_on:
-      - stitching-db
-      - dm-store
-    links:
-      - stitching-db
-      - dm-store
-    ports:
-      - 4630:8080
+#  stitching-api:
```
Comment (linusnorton) [inline docker-compose-dependencies.yml]: why is this commented out now?

File: docker-compose-dependencies.yml [hunk 3]
```diff
@@ -1,26 +1,31 @@
 ---
-version: '2.1'
+version: '3'
 
 services:
-  service-auth-provider-app:
+  service-auth-provider-api:
```
Comment (linusnorton) [inline docker-compose-dependencies.yml]: when did this change?

File: infrastructure/prod.tfvars [hunk 4]
```diff
@@ -1,3 +1,3 @@
 idam_api_base_uri = "https://idam-api.platform.hmcts.net"
-capacity = "2"
```
Comment (linusnorton) [inline infrastructure/prod.tfvars]: we still need this

File: infrastructure/sprod.tfvars [hunk 5]
```diff
@@ -1,2 +1,3 @@
 idam_api_base_uri = "http://idam-api-idam-sprod.service.core-compute-idam-sprod.internal"
 capacity = "2"
+enable_stitching_complete_callback="false"
```
Comment (linusnorton) [inline infrastructure/sprod.tfvars]: this file can go now

File: src/aat/java/uk/gov/hmcts/reform/em/orchestrator/performance/NpaLoad.scala [hunk 6]
```diff
@@ -15,7 +15,7 @@ class NpaLoad extends Simulation with HttpConfiguration {
 //    var jsonObject = new JSONObject().put("bundle", testUtil.getDocumentId).toString()
 //    def run = exec(
 //          http("create-npa-task-" + testConfig.pdfName)
-//            .post(Env.getTestUrl + "/api/document-tasks")
+//            .startStitchingTask(Env.getTestUrl + "/api/document-tasks")
```
Comment (linusnorton) [inline src/aat/java/uk/gov/hmcts/reform/em/orchestrator/performance/NpaLoad.scala]: We should kill all the old scala load tests - I've made a task for it: https://tools.hmcts.net/jira/browse/EM-2116

File: src/main/java/uk/gov/hmcts/reform/em/orchestrator/automatedbundling/configuration/LocalConfigurationLoader.java [hunk 7]
```diff
@@ -30,8 +30,7 @@ public BundleConfiguration load(String filename) {
 
         try {
             return mapper.readValue(input, BundleConfiguration.class);
-        }
-        catch (IOException e) {
+        } catch (IOException e) {
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/reform/em/orchestrator/automatedbundling/configuration/LocalConfigurationLoader.java]: Sorry, that's me... I just can't break the habit.

File: src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/ccdapi/CcdDataApiCaseUpdater.java [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+     * @param ccdCallbackDto - callbackDTO
+     * @param jwt - authentication
+     */
+    public void executeUpdate(CcdCallbackDto ccdCallbackDto, String jwt) {
+        try {
+            final String json = objectMapper.writeValueAsString(createCcdCaseDataContent(ccdCallbackDto));
+            final RequestBody body = RequestBody.create(json, MediaType.get("application/json"));
+            final Request updateRequest = new Request.Builder()
+                    .addHeader("Authorization", jwt)
+                    .addHeader("experimental", "true")
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/ccdapi/CcdDataApiCaseUpdater.java]: what does this do?

File: src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/ccdcallbackhandler/CcdEvent.java [hunk 9]
```diff
@@ -0,0 +1,21 @@
+package uk.gov.hmcts.reform.em.orchestrator.service.ccdcallbackhandler;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.NoArgsConstructor;
+import lombok.ToString;
+
+@ToString
```
Comment (linusnorton) [inline src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/ccdcallbackhandler/CcdEvent.java]: I wasn't aware we're using lombok on this repo... we aren't already using it I'd prefer not to start.

File: docker-compose-dependencies.yml (hunk 2 above)
Comment (pawelkaczmarekuk) [reply docker-compose-dependencies.yml]: newest stitching API is not created yet (in PR). I had to run it locally not via docker

File: src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/ccdapi/CcdDataApiCaseUpdater.java (hunk 8 above)
Comment (pawelkaczmarekuk) [reply src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/ccdapi/CcdDataApiCaseUpdater.java]: It allows to use v2 version of the API - Mario said it's safe

File: infrastructure/prod.tfvars (hunk 4 above)
Comment (pawelkaczmarekuk) [reply infrastructure/prod.tfvars]: let's solve it with a separate PR

File: docker-compose-dependencies.yml (hunk 3 above)
Comment (pawelkaczmarekuk) [reply docker-compose-dependencies.yml]: it's just a docker label

File: .env (hunk 1 above)
Comment (pawelkaczmarekuk) [reply .env]: that's needed by shared ccd docker database

File: src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/ccdcallbackhandler/CcdEvent.java (hunk 9 above)
Comment (pawelkaczmarekuk) [reply src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/ccdcallbackhandler/CcdEvent.java]: it's so much easier to write rapidly and concisely -

File: docker-compose-dependencies.yml (hunk 2 above)
Comment (linusnorton) [reply docker-compose-dependencies.yml]: this change shouldn't be merged to master

File: src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/ccdcallbackhandler/CcdEvent.java (hunk 9 above)
Comment (linusnorton) [reply src/main/java/uk/gov/hmcts/reform/em/orchestrator/service/ccdcallbackhandler/CcdEvent.java]: I'd prefer not to use magic but we can do

File: docker-compose-dependencies.yml (hunk 2 above)
Comment (pawelkaczmarekuk) [reply docker-compose-dependencies.yml]: agreed. I think it's best to merge stitching first and then uncomment that section.

**Pull request merged**